cEP-0024.md

```diff
@@ -0,0 +1,85 @@
+# Implement GitMate automations and plugins for coala
+
+| Metadata |                                                       |
+| -------- | ----------------------------------------------------- |
+| cEP      | 24                                                    |
+| Version  | 1.0                                                   |
+| Title    | Implement Gitmate automations and plugins for coala   |
+| Authors  | Vamshi Krishna Bommerla <mailto:bommvams72@gmail.com> |
+| Status   | Proposed                                              |
+| Type     | Process                                               |
+
+## Abstract
+
+This cEP describes the details of the process of creating new automations and
+plugins in GitMate which helps coala development workflow, as a part of the
+[GSoC project](https://summerofcode.withgoogle.com/projects/#5344316227059712).
+
+## Introduction
+
+[GitMate](https://gitmate.io/) is an automation tool for developers. While it
+works well, there area number of plugins which saves a lot of time of our
+community(for example automatically close PRs from master). Many of them are
+general plugins, which are required by every org, but we need more plugins
+which support coala development workflow.
+
+## New Features
+
+### Auto-assign issue based on commit mention
+
+When a commit mentions(fixes, closes) a unassigned issues, auto assign
```

Comment:
I am working on IGitt implementation(including GitLab)

Reply:
If the commit is from the fork of a coala repo, what we can do is listen for webhooks from pull requests which would contain the commit messages that can be filtered for mentions.

Reply:
I see that this was roughly where the "PR only" solution for this plugin originated.

listening to PRs only was not the intention of this project!

The idea was to start tracking work/workers before their first PR has been created, and even before they have been invited to the organisation, as most people invited to the org never do anything useful.

This plugin wasnt described in the "ideas" section of http://projects.coala.io/#/projects?project=implement_gitmate_plugins_for_coala 

Unfortunately the project page only mentioned "Implement auto-assign issue based on commit mention." in phase 1, and the gdoc proposal didnt contain more detail either.

So I created https://github.com/coala/projects/issues/532 with a little more context , so it could be added to the project ideas section  , and that hasnt been worked on yet.

There is still value in a plugin which helps with managing/tracking/assigning PRs of unassigned issues, but that misses the workflow improvement that was intended to reduce maintainer workload.

If IGitt does not support exposing the 'commit mentions' which GitHub and GitLab both add to an issue whenever a commit anywhere occurs (not i.e PRs), I think that should be implemented before this plugin is attempted.